### PR TITLE
Force retune

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Add a trigger to force restart of the tuning process upon the next call to `iteratePairwise()`

## ~Resolved Issues~

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] New Tests (`testForceRetuneInPhase` and `testForceRetuneBetweenPhases`=

### CODE REVIEW

File: src/autopas/selectors/AutoTuner.h [hunk 1]
```diff
@@ -149,7 +154,7 @@ class AutoTuner {
     if (_tuningStrategy->searchSpaceIsTrivial()) {
       return false;
     }
-
+    // TODO This will not return true if forceRetune is called during a tuning phase while sampling something.
```
Comment (SteffenSeckler) [inline src/autopas/selectors/AutoTuner.h:157]: todo?

**Review by SteffenSeckler — APPROVED** [review]

File: src/autopas/selectors/AutoTuner.h (hunk 1 above)
Comment (FG-TUM) [reply src/autopas/selectors/AutoTuner.h:157]: wait this is already fixed via 1f852ca :)

**Comment by SteffenSeckler** [comment]
fyi: I think that you are already hitting false-positive warnings in oxygen xD

**Comment by SteffenSeckler** [comment]
do the doxygen warnings vanish, after #596 is merged?

**Comment by FG-TUM** [comment]
> do the doxygen warnings vanish, after #596 is merged?

yes. They come from #593 , are currently part of the master and already correctly parsed in #596.

**Pull request merged**